### ISSUE TITLE
Sparse matrix not assigned to variable X

### ISSUE THREAD

**Issue by depkchowdary** [spyder-ide/spyder#5038]
## Description of your problem

Below is basic NLP classifier. when trying to create sparse matrix and assign it to variable X. Its not getting created. However, If I change X to any variable such as Inde_X it is displayed in variable explorer. 

PFB code I am running. 

`import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

#import any dataset
dataset = pd.read_csv('Restaurant_Reviews.tsv', delimiter = '\t', quoting = 3)

#clean data befor using NLP
import re
import nltk
nltk.download('stopwords')
from nltk.corpus import stopwords
from nltk.stem.porter import PorterStemmer
ps = PorterStemmer()
corpus = []
for i in range(0,1000):
    review = re.sub('[^A-Za-z]',' ', dataset['Review'][i])
    review= review.lower()
    review = review.split()
    review = [ps.stem(word) for word in review if not word in set(stopwords.words('English'))]
    review = " ".join(review)
    corpus.append(review)
# Creating the Bag of Words model
from sklearn.feature_extraction.text import CountVectorizer
cv = CountVectorizer(max_features = 1500)
#The array is returned but not getting assigned to X
X = cv.fit_transform(corpus).toarray()
y = dataset.iloc[:, 1].values
`

**What steps will reproduce the problem?**

1. Run the code given
2. Look for X in variable explorer
3. X is not created

**What is the expected output? What do you see instead?**
X should be created and assigned a sparse matrix returned by scikit learn CountVectorizer function.

**Please provide any additional information below**

## Versions and main components

* Spyder Version: 3.1.4 and 3.2.1
* Python Version: 3.6
* Qt Version:
* PyQt Version:
* Operating system: windows 7

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:
IPython >=4.0    :  5.3.0 (OK)
cython >=0.21    :  0.25.2 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
numpy >=1.7      :  1.12.1 (OK)
pandas >=0.13.1  :  0.20.1 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.5.6 (OK)
sympy >=0.7.3    :  1.0 (OK)

**Comment by depkchowdary** [comment]
Same goes for X_test and X_train variable 

`
#nlp classifier
# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(indie_X, y, test_size = 0.20, random_state = 0)
`

**Comment by ccordoba12** [comment]
Variables that start with capital letters are hidden by default. You can show them by going to Variable Explorer options menu (the little cog to its right) and deselecting the option called

    Exclude capitalized references
